public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -384,6 +385,22 @@ export class LokiDatasource
     return res.data || [];
   }
 
+  async queryStatsRequest(query: LokiQuery): Promise<QueryStats> {
+    const labelMatcher = `{${query.expr.split('{')[1].split('}')[0]}}`;
+    const { start, end } = this.getTimeRangeParams();
+
+    const { data } = await lastValueFrom(
```

Comment:
What do we do if this request fails, i.e. in case of an older Loki version?

Reply:
@svennergr there are two possible ways for this to fail.

**First:**
the user is using a version that supports this endpoint, however they are not configured to use `tsdb`.
in this case we still make the request but loki will always return `{ streams: 0, chunks: 0, bytes: 0, entries: 0 }`. 
we have a check to catch this and if the value of `queryStats.bytes` is falsy we just don’t display the tooltip.

**Second:**
the user is using a version of loki where this endpoint is not supported
in this case loki returns a 404 so `queryStats.bytes` is falsy, meaning it it caught by the same logic as above and the tooltip is not displayed.

this was the easiest solution as we have no way to detect the loki version being used (from my knowledge, correct me if i am wrong).